doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1259,18 +1259,18 @@ Managing the local context
 
 .. tacn:: clear {? {? - } {+ @ident } }
 
-   Erases hypotheses from the context of the current goal, provided that the
-   remaining hypotheses and the goal don't depend directly or indirectly on the
-   erased hypotheses.
+   Erases *unneeded* hypotheses from the context of the current goal.  "Unneeded"
+   means that the unselected hypotheses and the goal don't depend directly or
```

Comment:
I wanted to distinguish selected and erased because they are different notions.
"remaining" meaning remaining after what?  After erasure?  That's a circular definition.

I changed this one:

```
   :n:`clear - {+ @ident }`
     Selects all hypotheses that are not named by the :n:`@ident`\s, then
     erases those that are unneeded.
```

OK?

Reply:
IMHO this formulation is complex and it is not completely obvious what is meant. OTOH, I don't have more ideas to propose so maybe let's keep what you did.

Reply:
The double negation ("not named" and "unneeded") make a harder to understand than usual, but I think it is correct and precise.  I suppose one could add an example, but that I think that's too much.